src/k8s/pkg/k8sd/features/localpv/localpv_test.go

```diff
@@ -0,0 +1,155 @@
+package localpv_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+
+	"github.com/canonical/k8s/pkg/client/helm"
+	helmmock "github.com/canonical/k8s/pkg/client/helm/mock"
+	"github.com/canonical/k8s/pkg/k8sd/features/localpv"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	snapmock "github.com/canonical/k8s/pkg/snap/mock"
+)
+
+func TestDisabled(t *testing.T) {
+	t.Run("HelmApplyFails", func(t *testing.T) {
+		g := NewWithT(t)
+
+		applyErr := errors.New("failed to apply")
+		helmM := &helmmock.Mock{
+			ApplyErr: applyErr,
+		}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+			},
+		}
+		cfg := types.LocalStorage{
+			Enabled:       ptr.To(false),
+			Default:       ptr.To(true),
+			ReclaimPolicy: ptr.To("reclaim-policy"),
+			LocalPath:     ptr.To("local-path"),
+		}
+
+		status, err := localpv.ApplyLocalStorage(context.Background(), snapM, cfg, nil)
+
+		g.Expect(err).To(MatchError(applyErr))
+		g.Expect(status.Enabled).To(BeFalse())
+		g.Expect(status.Message).To(ContainSubstring(applyErr.Error()))
```

Comment:
It's possible but I think this check is more in the "implementation-testing" realm than in the "behavior-testing" one. I believe adding these kinds of checks will mostly introduce headaches on future UX changes. wdyt?

Reply:
Generally, testing for error messages is a fragile construct that should be avoided. 
Testing the whole error would break whenever *anything* in the whole chain changes
Testing a substring is better but still relies on subtle implementation details. 

You could either have custom errors where needed (IMHO not here) or simply test for the existence of the error.

Reply:
I'm saying this because if we decided to change a single word in the returned message in the future, we need to change it twice, once in the actual implementation and once in the tests. and I think that this specific check does not introduce any further robustness and confidence.